### PR TITLE
PN-10587-fe-api-institutions-and-products-pa

### PR DESCRIPTION
**Pull request by ImJimMoriarty**
## Short Description
Migrate the old Institutions and Products API to the BFF's API.

## List of Changes Proposed in This Pull Request
- Generated models from the BFF OpenAPI.
- Updated `getInstitutions()` and `getInstitutionsProducts()` with the BFF's API.
- Updated related tests.
- Updated related mocks.

## How to Test
Start PA and verify if you can see the list of institutions and products on the dashboard.

### CODE REVIEW

File: packages/pn-pa-webapp/src/api/external-registries/External-registries.api.ts [hunk 1]
```diff
@@ -1,16 +1,8 @@
 import { AxiosResponse } from 'axios';
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/api/external-registries/External-registries.api.ts:1]: Remove the file

File: packages/pn-pa-webapp/src/__test__/App.test.tsx [hunk 2]
```diff
@@ -96,8 +92,10 @@ describe('App', async () => {
 
   it('render component - user logged in', async () => {
     mock.onGet('/bff/v1/tos-privacy').reply(200, tosPrivacyConsentMock(true, true));
-    mock.onGet(GET_INSTITUTIONS()).reply(200, institutionsList);
-    mock.onGet(GET_INSTITUTION_PRODUCTS('1')).reply(200, productsList);
+    mock.onGet('webapi.dev.notifichedigitali.it/bff/v1/institutions').reply(200, institutionsDTO);
+    mock
+      .onGet('webapi.dev.notifichedigitali.it/bff/v1/institutions/products')
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/__test__/App.test.tsx:97]: use relative path here and below

File: packages/pn-pa-webapp/src/api/external-registries/__test__/External-registries.api.test.ts [hunk 3]
```diff
@@ -1,20 +1,10 @@
 import MockAdapter from 'axios-mock-adapter';
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/api/external-registries/__test__/External-registries.api.test.ts:1]: remove the file

File: packages/pn-pa-webapp/src/api/external-registries/__test__/external-registries-routes.test.ts [hunk 4]
```diff
@@ -1,18 +1,8 @@
-import { GET_PARTY_FOR_ORGANIZATION, GET_INSTITUTIONS, GET_INSTITUTION_PRODUCTS } from "../external-registries-routes";
+import { GET_PARTY_FOR_ORGANIZATION } from '../external-registries-routes';
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/api/external-registries/__test__/external-registries-routes.test.ts:1]: remove the file

File: packages/pn-pa-webapp/src/api/external-registries/external-registries-routes.ts [hunk 5]
```diff
@@ -7,16 +7,12 @@ const API_EXTERNAL_REGISTRIES_PREFIX = 'ext-registry';
 const PA_SEGMENT = 'pa';
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/api/external-registries/external-registries-routes.ts:7]: remove the file

File: packages/pn-pa-webapp/src/generated-client/institution-and-product/.openapi-generator-ignore [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+# You can recursively match patterns against a directory, file or extension with a double asterisk (**):
+#foo/**/qux
+# This matches foo/bar/qux, foo/baz/qux, and foo/bar/baz/qux
+
+# You can also negate patterns with an exclamation (!).
+# For example, you can ignore all files in a docs folder with the file extension .md:
+#docs/*.md
+# Then explicitly reverse the ignore rule for a single file:
+#!docs/README.md
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/generated-client/institution-and-product/.openapi-generator-ignore:23]: check the same file on the others generated directory and add the additional data

File: packages/pn-pa-webapp/src/__mocks__/User.mock.ts [hunk 7]
```diff
@@ -1,49 +1,34 @@
-import { PartyEntity, ProductEntity } from '@pagopa/mui-italia';
+import { PartyEntityWithUrl } from "@pagopa-pn/pn-commons";
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/__mocks__/User.mock.ts:1]: Why the use of PartyEntityWithUrl and not PartyEntity?

File: packages/pn-personafisica-webapp/openapitools.json [hunk 8]
```diff
@@ -25,6 +25,17 @@
           "modelPackage": "models",
           "cleanupOutput": true
         }
+      },
+      "bff-institution-and-product": {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/openapitools.json:29]: In pf we don't have the institution-and-product

File: packages/pn-personagiuridica-webapp/openapitools.json [hunk 9]
```diff
@@ -25,6 +25,17 @@
           "modelPackage": "models",
           "cleanupOutput": true
         }
+      },
+      "bff-institution-and-product": {
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/openapitools.json:29]: In pg we don't have the institution-and-product

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-pa-webapp/src/__mocks__/User.mock.ts (hunk 7 above)
Comment (ImJimMoriarty) [reply packages/pn-pa-webapp/src/__mocks__/User.mock.ts:1]: Because there isn't the attribute `entityUrl` and this is what the real api should return

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-pa-webapp/src/__test__/App.test.tsx [hunk 10]
```diff
@@ -96,8 +92,8 @@ describe('App', async () => {
 
   it('render component - user logged in', async () => {
     mock.onGet('/bff/v1/tos-privacy').reply(200, tosPrivacyConsentMock(true, true));
-    mock.onGet(GET_INSTITUTIONS()).reply(200, institutionsList);
-    mock.onGet(GET_INSTITUTION_PRODUCTS('1')).reply(200, productsList);
+    mock.onGet('/v1/institutions').reply(200, institutionsDTO);
```
[the remark this answers is not in the document]
Comment (ImJimMoriarty) [reply packages/pn-pa-webapp/src/__test__/App.test.tsx:95]: I tried but it doesn't work with

**Review by AndreaCimini90 — APPROVED** [review]

**Pull request merged**